### PR TITLE
[Tests] Start adding tests for internal helpers & migrate from c8 → v8 coverage reporter

### PR DESCRIPTION
**Pull request by Vpr99**
tl;dr: I think we can merge this whenever. I'll just keep writing tests as time permits 😄 

---

This PR makes some tooling changes to vitest and starts adding tests to the internal helper functions.

### Tests

- `$lib/internal/helpers/array`
  - [X] `last`
  - [X] `next`
  - [X] `prev`
  - [X] `wrapArray`
- `$lib/internal/helpers/callbacks`
  - [X] `executeCallbacks`
- `$lib/internal/helpers/dom`
  - [X] `focus`
  - [X] `getTabbableNodes`
- `$lib/internal/helpers/list`
  - [X] `getFirstOption`
  - [X] `getOptions`
- `$lib/internal/helpers/object`
  - [X] `omit`
- `$lib/internal/helpers/keyboard`
  - [X] `getNextKey`
  - [X] `getPrevKey`
  - [X] `getDirectionalKeys`

### Misc. other stuff

- The focusTrap used a `toReadable` helper which converts a writable store to a readable one. This functionality is built into Svelte and exposed via the `readonly` function (https://svelte.dev/docs/svelte-store#readonly). The bespoke helper has been removed from focusTrap in favor of the built-in.
- `$lib/internal/helpers/numbers` was deleted because it only contained one utility (`clamp`) which wasn't used.
- `objectEntries`, `objectKeys`, and `joinKeys` object utilities were deleted since they're unused.
- The `getPageItems` test has also been rewritten to use [`test.each`](https://vitest.dev/api/#test-each) syntax which is part of the Vitest API.

---

### Coverage reporter

When running a coverage report, I got a warning from Vitest that the c8 coverage reporter was being deprecated and replaced by v8. This was corroborated by the [Coverage](https://vitest.dev/guide/coverage.html#coverage) section of their docs. I replaced `@vitest/coverage-c8` with `@vitest/coverage-v8` and added a coverage script.

### CODE REVIEW

File: src/lib/builders/pagination/tests/index.spec.ts [hunk 1]
```diff
@@ -1,56 +1,59 @@
...
+		[{ totalPages: 7, page: 4, siblingCount: undefined, expected: '1,...,3,4,5,...,7' }],
+		[{ totalPages: 3, page: 2, siblingCount: undefined, expected: '1,2,3' }],
+		[{ totalPages: 4, page: 2, siblingCount: undefined, expected: '1,2,3,4' }],
+		[{ totalPages: 100, page: undefined, siblingCount: 2, expected: '1,2,3,4,5,...,100' }],
+		[{ totalPages: 100, page: undefined, siblingCount: 3, expected: '1,2,3,4,5,6,...,100' }],
+		[{ totalPages: 100, page: 5, siblingCount: 3, expected: '1,2,3,4,5,6,...,100' }],
+		[{ totalPages: 100, page: 6, siblingCount: 3, expected: '1,...,3,4,5,6,7,8,9,...,100' }],
+		[{ totalPages: 100, page: 97, siblingCount: 3, expected: '1,...,95,96,97,98,99,100' }],
+		[{ totalPages: 100, page: 96, siblingCount: 3, expected: '1,...,95,96,97,98,99,100' }],
+		[
```
Comment (Vpr99) [inline src/lib/builders/pagination/tests/index.spec.ts:38]: Sucks that the formatting for this one test case doesn't inline 😢

Comment (Vpr99) [reply src/lib/builders/pagination/tests/index.spec.ts:39]: Eh I'll leave it alone—it just offends my sensibilities hahaha. Thanks :)

**Comment by huntabyte** [comment]
Nice work! This looks good to me! I'll let @TGlide make the call on whether we add changesets to internal changes or not!

File: .changeset/soft-hotels-own.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+'@melt-ui/svelte': patch
+---
+
+[Tooling] Migrate coverage reporter from c8 to v8
+[Tests] Add tests for internal helpers:
```
Comment (TGlide) [inline .changeset/soft-hotels-own.md:6]: No need to update version if we're just adding tests, you can delete this file :)

File: src/lib/internal/helpers/tests/array.spec copy.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { describe, expect, test } from 'vitest';
```
Comment (TGlide) [inline src/lib/internal/helpers/tests/array.spec copy.ts:1]: The naming of this file is weird, no?

File: src/lib/internal/helpers/tests/array.spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { describe, expect, test } from 'vitest';
```
Comment (TGlide) [inline src/lib/internal/helpers/tests/array.spec.ts:1]: Oh it's duplicated 😄

**Review by TGlide — CHANGES_REQUESTED** [review]
Hey, thanks for this, this is awesome!!! Just some small changes needed 😄

File: src/lib/internal/helpers/tests/array.spec.ts (hunk 4 above)
Comment (Vpr99) [reply src/lib/internal/helpers/tests/array.spec.ts:1]: Ahh yep great catch! I deleted the dupe :)

File: .changeset/soft-hotels-own.md (hunk 2 above)
Comment (Vpr99) [reply .changeset/soft-hotels-own.md]: Done!

**Review by TGlide — APPROVED** [review]
